### PR TITLE
Sample for Layer View State

### PR DESCRIPTION
**Pull request by apurvag123**

### CODE REVIEW

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            UIStackView stackView = new UIStackView(new CoreGraphics.CGRect(0, 0, View.Bounds.Width, View.Bounds.Height))
+            {
+                Axis = UILayoutConstraintAxis.Vertical,
+                Alignment = UIStackViewAlignment.Fill,
+                Distribution = UIStackViewDistribution.FillProportionally,
+                BackgroundColor = UIColor.Gray
+            };
+
+            // Setup the visual frame for the MapView
+            _myMapView = new MapView()
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Should this take into account top bar height?

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            _myMapView.LayerViewStateChanged += OnLayerViewStateChanged;
+
+            // Provide used Map to the MapView
+            _myMapView.Map = myMap;
+        }
+
+        private void OnLayerViewStateChanged(object sender, LayerViewState e)
+        {
+            // State changed event is sent by a layer. In the list, find the layer which sends this event. If it exists then update the status
+            LayerStatusModel model = _layerStatusModels.Where(l => l.LayerName == (sender as Layer).Name).FirstOrDefault();
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Please split into to 2 lines. 
1) cast sender to variable
2) Where clause

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+            // Set the visible scale range for the image layer
+            imageLayer.MinScale = 40000000;
+            imageLayer.MaxScale = 2000000;
+
+            // Add the image layer to map
+            myMap.OperationalLayers.Add(imageLayer);
+
+            //Create a feature layer using url
+            FeatureLayer featureLayer = new FeatureLayer(new Uri("http://sampleserver6.arcgisonline.com/arcgis/rest/services/Recreation/FeatureServer/0"));
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Please split into 2 lines.

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+            // Create a tiled layer using url
+            ArcGISTiledLayer tiledLayer = new ArcGISTiledLayer(new Uri("http://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer"));
+            tiledLayer.Name = "Tiled Layer";
+
+            // Add the tiled layer to map
+            myMap.OperationalLayers.Add(tiledLayer);
+
+            // Create ArcGISMapImage layer using a url
+            ArcGISMapImageLayer imageLayer = new ArcGISMapImageLayer(new Uri("http://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer"));
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Please split into 2 lines.

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // Create new Map
+            Map myMap = new Map();
+
+            // Create a tiled layer using url
+            ArcGISTiledLayer tiledLayer = new ArcGISTiledLayer(new Uri("http://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer"));
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Please split into 2 lines.

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 6]
```diff
@@ -0,0 +1,199 @@
+using System;
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Add copyright information

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+using System.Drawing;
+
+using CoreFoundation;
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Reorder usings to alphabetical and remove non used ones. Please use the refactoring provided by the Visual Studio.

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+            //Create a feature layer using url
+            FeatureLayer featureLayer = new FeatureLayer(new Uri("http://sampleserver6.arcgisonline.com/arcgis/rest/services/Recreation/FeatureServer/0"));
+            featureLayer.Name = "Feature Layer";
+
+            // Add the feature layer to map
+            myMap.OperationalLayers.Add(featureLayer);
+
+            // Create a mappoint the map should zoom to
+            MapPoint mapPoint = new MapPoint(-11000000, 4500000, SpatialReferences.WebMercator);
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: please do not use class name as a variable name.

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs (hunk 3 above)
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: please do not use class name as a variable name.

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json [hunk 9]
```diff
@@ -0,0 +1,14 @@
+{
+  "Name": "LayerView State",
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json]: use : Display layer view state

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+        // Create and hold reference to tableview
+        private UITableView _tableView;
+
+        // Reference to list of view status for each layer
+        private List<LayerStatusModel> _layerStatusModels = new List<LayerStatusModel>();
+
+        public DisplayLayerViewState()
+        {
+            this.Title = "Display LayerView State";
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: Use Display layer view state

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  "SampleName": "DisplayLayerViewState",
+  "Description": "This sample demonstrates how to get view status for layers in a map.",
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json]: Please refer https://developers.arcgis.com/ios/beta/swift/sample-code/display-layer-view-state.htm for the description.

**Comment by anttikajanus** [comment]
@apurvag123 Reviewed, back to you.

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs (hunk 10 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: done

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json (hunk 11 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json]: This is same as https://developers.arcgis.com/ios/beta/swift/sample-code/display-layer-view-state.htm

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json (hunk 9 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/metadata.json]: done

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs (hunk 8 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: done

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs (hunk 6 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: done

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs (hunk 7 above)
Comment (apurvag123) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: done

**Comment by apurvag123** [comment]
Made the changes

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs [hunk 12]
```diff
@@ -91,7 +106,11 @@ private void Initialize()
         private void OnLayerViewStateChanged(object sender, LayerViewState e)
         {
             // State changed event is sent by a layer. In the list, find the layer which sends this event. If it exists then update the status
-            LayerStatusModel model = _layerStatusModels.Where(l => l.LayerName == (sender as Layer).Name).FirstOrDefault();
+            Layer myLayer = sender as Layer;
+
+            String layerName = myLayer.Name;
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: No need to split this to another line. so 1 line for getting the layer and then doing the query. Main reason for this is to keep the where part easy to read. When we nest multiple () inside of one lambda it gets hard to read. 

You should use `var` with the types that aren't Esri types and qualify types when using our types. This is done to make it very clear what part of the types are ours and what aren't.

````CSharp
var myLayer = sender as Layer;
var model = _layerStatusModels.Where(model => model.LayerName == myLayer.name).FirstOrDefault(); 
````

File: src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs (hunk 1 above)
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/MapView/DisplayLayerViewState/DisplayLayerViewState.cs]: How about this?

**Comment by apurvag123** [comment]
Is it okay to merge this?

**Comment by anttikajanus** [comment]
Please merge.

**Pull request merged**